python/ray/experimental/sgd/interaces.py

```diff
@@ -0,0 +1,134 @@
+from dataclasses import dataclass
+
+from typing import Callable
+
+
+@dataclass
+class WorkerArgs:
+    num_workers: int
+    num_cpus_per_worker: int
+    num_gpus_per_worker: int
+    placement_strategy: str # Assumes 1 bundle per worker.
+
+    elastic_training: bool = False # If True, training will continue with fewer
+    # workers if an actor fails. Also, if more resources are available,
+    # training will continue with more workers.
+    max_actor_restarts: int = 0 # Number of retries when Ray actor fails.
+    # Doesn't do anything if elastic_training is True. Set to -1 for
+    # unlimited restarts.
+    max_failed_actors: int = 0 # If `elastic_training` is True, this
+            # specifies the maximum number of failed actors with which
+            # we still continue training.
+
+class BackendConfig:
+    pass
+
+@dataclass
+class TorchConfig(BackendConfig):
+    address: str # The connection address to use for the process group.
+    process_group_timeout: int # The timeout for process group creation.
+
+@dataclass
+class TFConfig(BackendConfig):
+    pass
+
+
+def setup_torch_process_group(...):
+    pass
+
+def shutdown_torch(...):
+    pass
+
+class TorchExecutor:
+    def __init__(self, worker_args: WorkerArgs, backend_config: TorchConfig,
+                 initialization_hook):
+        self.actor_group = ActorGroup(num_workers=worker_args.num_workers,
+                                  cpus_per_worker=worker_args.num_cpus_per_worker,
+                                  gpus_per_worker=worker_args.num_gpus_per_worker,
+                                  pg_strategy=worker_args.placement_strategy)
+        self.backend_config = backend_config
+        self.initialization_hook = initialization_hook
+
+    def start(self):
+        self.actor_group.start()
+
+        self.actor_group.execute(self.initialization_hook)
+
+        futures = []
+        for i in range(len(self.actor_group.workers)):
+            futures.append(self.actor_group.execute_single_async(i,
+                                                  setup_torch_process_group,
+                                            kwargs={
+                "address": self.backend_config.address,
+                "process_group_timeout": self.backend_config.process_group_timeout,
+                "world_size": len(self.actor_group.workers),
+                "rank": i
+            }))
+
+        ray.get(futures)
+
+    def run(self, training_fn, config):
```

Comment:
Rename this to `execute` so that we are consistent with the pattern of `run` meaning `start -> execute -> shutdown`?

Reply:
yep sounds good